migrations/2022-05-11-123144_language-tags/up.sql

```diff
@@ -0,0 +1,4 @@
+alter table local_user rename column lang to interface_language;
+alter table local_user add column discussion_languages varchar(3)[] not null default '{}';
```

Comment:
I dont think a table like `local_user_discussion_language(id, local_user_id, lang)` makes sense, as we are dealing with a fixed list of languages, so it would just waste space. Instead i would make a table like `all_discussion_languages(id, lang)`, and populate that with a code migration.

Yes, `und` stands for "undetermined language", and is part of [ISO_639-2](https://en.wikipedia.org/wiki/ISO_639-2#Special_situations).

Reply:
Thinking about this more, we should really have a `language` table: (id, language_code)`, and both the local_user_discussion_language, and the post table, should reference the `lang_id`, so that joins are more optimized.

Then we can do proper foreign-key constraints, and the joins should be fast.